File: python/pages/upload_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class UploadPage(BasePage):

    FILE_INPUT = (By.ID, "file-upload")
    UPLOAD_BTN = (By.ID, "upload-btn")
    RESULT = (By.ID, "upload-result")

    def upload_file(self, path):
        self.upload_file(self.FILE_INPUT, path)
        self.click(self.UPLOAD_BTN)
        return self

    def get_result(self):
        return self.get_text(self.RESULT)

File: src/test/java/com/example/pages/UploadPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class UploadPage extends SeleniumBasePage {

    private final By fileInput = By.id("file-upload");
    private final By uploadButton = By.id("upload-btn");
    private final By result = By.id("upload-result");

    public UploadPage(WebDriver driver) {
        super(driver);
    }

    public UploadPage uploadFile(String filePath) {
        uploadFile(fileInput, filePath);
        click(uploadButton);
        return this;
    }

    public String getResult() {
        return getText(result);
    }
}
